### PR TITLE
docs(requirements): Add client requirements document - Ticket #21

### PR DESCRIPTION
**Pull request by wolfgabo**
Here is the document

### CODE REVIEW

File: VIKINGOS-ANALYZER-V1.0.md [hunk 1]
```diff
@@ -0,0 +1,337 @@
+#VIKINGOS ANALYZER V1.0
```
Comment (CarlosMoriel) [inline VIKINGOS-ANALYZER-V1.0.md:1]: markdown formatting does not display correctly, please use this page to validate your preview.

https://www.digitalocean.com/community/markdown?md=

Tip: When you add a # please add a space to see the title correctly
example:
"#VIKINGOS ANALYZER V1.0" ❌
"# VIKINGOS ANALYZER V1.0" ✅

**Review by CarlosMoriel — CHANGES_REQUESTED** [review]

File: VIKINGOS-ANALYZER-V1.0.md [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+
+[REQUIREMENTS](https://github.com/EstandaresDeCalidadSW23/vikingos-analyser/issues/21#event-10289307088%20%22REQUIREMENTS%22 "REQUIREMENTS")
+
+##• Define the MVP product
```
Comment (CarlosMoriel) [inline VIKINGOS-ANALYZER-V1.0.md:5]: ## Define the MVP product

Comment (CarlosMoriel) [inline VIKINGOS-ANALYZER-V1.0.md:5]:
```suggestion
##  Define the MVP product
```

File: VIKINGOS-ANALYZER-V1.0.md [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+     Minimal: Contains only the essential features necessary to solve a problem or satisfy a specific user need. All non-critical functionality is removed to simplify early development.
+     Viable: It is functional and usable. Although basic, the product must be effective enough to provide real value to users and fulfill its primary purpose.
+     Product: It can be software, a mobile application, a website, a physical device or any other type of product or service that a company is developing.
+A VARIABLE CANNOT BE USED WITHOUT BEING DECLARED[REQUIREMENTS](https://github.com/EstandaresDeCalidadSW23/vikingos-analyser/issues/21#event-10289307088 "REQUIREMENTS")
+variable
+my variable= 5; 0-1,000,000 (FROM ZERO TO A MILLION)
+
+-#Variable is declared
```
Comment (CarlosMoriel) [inline VIKINGOS-ANALYZER-V1.0.md:13]:
```suggestion
-# Variable is declared
```

File: VIKINGOS-ANALYZER-V1.0.md [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+
+     Behavior of data structures: Evaluates the behavior of data structures, such as arrays or lists, to ensure that they are used correctly and that there are no overflows, indexes out of range, etc.
+
+     Interactions with the environment: If the program interacts with the environment, such as operating systems, files, or databases, verify that these interactions are handled correctly and that there are no security issues or unauthorized access.
+
+     Optimization and performance: Evaluate the performance of the program to identify potential bottlenecks or inefficiencies in the code, and ensure it is optimized as needed.
+
+     Peer Review and Testing: Conduct extensive peer reviews and tests to verify the behavior and functionality of the program from a semantic perspective.
+
+##1.1. Product Features
```
Comment (CarlosMoriel) [inline VIKINGOS-ANALYZER-V1.0.md:102]:
```suggestion
## 1.1. Product Features
```

File: VIKINGOS-ANALYZER-V1.0.md [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+     Design the user experience:
+         Make sure the user experience is intuitive and easy to follow. Help content should be clearly organized and accessible, and users should be able to close the pop-up or help panel easily.
+
+     Tests and review:
+         Extensive testing to ensure that the help button works correctly in different situations and platforms. Make sure your help content is accurate and helpful to users.
+
+     Documentation and update:
+         Keep help content updated as the program evolves. Add new documentation or update existing documentation as needed.
+
+##•	EXECUTION
```
Comment (CarlosMoriel) [inline VIKINGOS-ANALYZER-V1.0.md:139]:
```suggestion
##	EXECUTION
```

File: VIKINGOS-ANALYZER-V1.0.md [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+     Deployment:
+         If the program runs in a production environment, it follows a controlled and documented deployment procedure.
+
+     Continuous monitoring:
+         After the initial run, establish a continuous monitoring system to detect problems and errors in real time.
+
+     Update Management:
+         Implements an update management process to apply security patches and software enhancements over time.
+
+##1.2. design constraints
```
Comment (CarlosMoriel) [inline VIKINGOS-ANALYZER-V1.0.md:185]:
```suggestion
## 1.2. design constraints
```

File: VIKINGOS-ANALYZER-V1.0.md [hunk 7]
```diff
@@ -0,0 +1,337 @@
...
+     Bug Management and Improvements:
+         Address bugs and issues identified during execution, and plan further improvements as needed.
+
+Example of a constraint:
+When the person makes a mistake, clearly show the mistake "and tell why they made a mistake"
+
+##3 Generate a backlog
```
Comment (CarlosMoriel) [inline VIKINGOS-ANALYZER-V1.0.md:192]:
```suggestion
## 3 Generate a backlog
```

File: VIKINGOS-ANALYZER-V1.0.md [hunk 8]
```diff
@@ -0,0 +1,337 @@
...
+
+     Update regularly: The backlog is not static; it evolves as more information is acquired and priorities change. Update the backlog on a regular basis based on user feedback, changes in requirements, and project progress.
+
+     Sprint planning:
+If you are using Scrum, select a set of user stories from the backlog to include in a sprint. The sprint backlog becomes the working set for the next sprint.
+
+     Monitoring and management:
+Use project management tools like Kanban boards or issue tracking software to track user stories and team progress.
+
+##4 BUGS
```
Comment (CarlosMoriel) [inline VIKINGOS-ANALYZER-V1.0.md:214]:
```suggestion
## 4 BUGS
```

File: VIKINGOS-ANALYZER-V1.0.md [hunk 9]
```diff
@@ -0,0 +1,337 @@
...
+
+     Safety tests:
+         Requirement: Perform regular security testing to detect and fix vulnerabilities that can be exploited by attackers.
+         
+         Benefit: Protects the software against possible threats and attacks.
+
+     Team Training:
+         Requirement: Provide ongoing training to the development team to keep them abreast of best practices and new technologies.
+
+##6 ENHANCEMENTS
```
Comment (CarlosMoriel) [inline VIKINGOS-ANALYZER-V1.0.md:263]:
```suggestion
## 6 ENHANCEMENTS
```

File: VIKINGOS-ANALYZER-V1.0.md [hunk 10]
```diff
@@ -0,0 +1,337 @@
...
+     Impact on existing system:
+         Evaluate and document how the improvement will affect the existing system. Will there be interactions with other functionalities or modules? Can there be negative impacts?
+
+     Test requirements:
+         Define clear acceptance criteria so that the development team can verify that the improvement has been implemented correctly.
+
+     Effort estimate:
+         Estimate the effort required to implement the improvement. This can include the time required, the resources involved, and any impact on existing deadlines.
+
+##8 TECH DEBT
```
Comment (CarlosMoriel) [inline VIKINGOS-ANALYZER-V1.0.md:282]:
```suggestion
## 8 TECH DEBT
```

File: VIKINGOS-ANALYZER-V1.0.md [hunk 11]
```diff
@@ -0,0 +1,337 @@
...
+
+     Follow-up and review:
+         Once technical debt has been addressed, perform regular monitoring and review to assess whether issues have been resolved and whether there have been improvements to the technical quality of the software.
+
+     continuing education:
+         Encourages continuous education of the development team to avoid accumulating new technical debt in the future. This may include training in good development practices.
+
+SPIKE research (they are light tasks either to conclude the topic or to say “I would like to implement this new prime voult”)
+
+##2. Specific requirements
```
Comment (CarlosMoriel) [inline VIKINGOS-ANALYZER-V1.0.md:318]:
```suggestion
## 2. Specific requirements
```

File: VIKINGOS-ANALYZER-V1.0.md [hunk 12]
```diff
@@ -0,0 +1,337 @@
...
+The following are the functional requirements that the system must implement. All requirements specified below are pending approval.
+Lexicon Program version 1.0
+The Lexicon program allows syntactic and semantic analysis.
+
+Requirements
+The description of the different requirements to be implemented in the software is written below.
+• The written message must not begin with numbers.
+• The numbers accepted by the systems must be between 0 - 1,000,000 (1 million), and these cannot be decimals, only whole numbers.
+
+##3. Documentation requirements
```
Comment (CarlosMoriel) [inline VIKINGOS-ANALYZER-V1.0.md:328]:
```suggestion
## 3. Documentation requirements
```

File: VIKINGOS-ANALYZER-V1.0.md [hunk 13]
```diff
@@ -0,0 +1,337 @@
...
+
+##3.1. User manual
```
Comment (CarlosMoriel) [inline VIKINGOS-ANALYZER-V1.0.md:330]:
```suggestion
## 3.1. User manual
```

File: VIKINGOS-ANALYZER-V1.0.md [hunk 14]
```diff
@@ -0,0 +1,337 @@
...
+A manual should be produced that specifies and exemplifies the main functionalities of the system. This manual will be a means by which users can learn to use the system, so it must be done for this purpose.
+
+##3.2. Online help
```
Comment (CarlosMoriel) [inline VIKINGOS-ANALYZER-V1.0.md:333]:
```suggestion
## 3.2. Online help
```

File: VIKINGOS-ANALYZER-V1.0.md [hunk 15]
```diff
@@ -0,0 +1,337 @@
...
+The client or user stated that they want any type of online help.
+
+##3.3. business rule
```
Comment (CarlosMoriel) [inline VIKINGOS-ANALYZER-V1.0.md:336]:
```suggestion
## 3.3. business rule
```

**Review by CarlosMoriel — CHANGES_REQUESTED** [review]

**Review by bcisneros — CHANGES_REQUESTED** [review]
This PR does not meet the Acceptance Criteria:

- The document should be inside `docs` folder
- Filename should be REQUIREMENTS.md. In this case, I liked the name you proposed, but I have the following suggestion: `docs/v1.0/REQUIREMENTS.md`
- The document is kind of difficult to read due to the following reasons:
   - Is not formatted properly
   - The structure is not clear, it is mixed with requirements and generic definitions of the requirements. I liked you added those but I don't think this is the correct place to put that content. Please keep this document simple.

My proposed structure is:

##  MVP
(Here you add the code example I proposed)
```
# This is the code we expect to validate

... put the code here
```
## General requirements
(Here you put all requirements related to the logic)
Example:
### G.1 Code must support comments
- Comments start with `#`
- Comments are ignored by the analyzer
...
### G.2 Variable declaration
- Only integer variables can be declared
- To declare a variable use the keyword `variable`
...
### G.3 Conditional statement
... do you follow the idea?

... and more sections

Note: I used the `G` prefix to indicate is a General requirement

## UI requirements
### UI.1 Line numbers
- The code editor must show the line numbers
... and continue with the other requirements

## Optional requirements
Describe other requirements that were defined as desirable but not mandatory

## Acceptance Tests
(I described 3 acceptance tests, please put them here)

**Pull request closed without merging**